src/Stratis.Features.FederatedPeg/SourceChain/RetrievalTypeConfirmations.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Features.FederatedPeg.Interfaces;
+
+namespace Stratis.Features.FederatedPeg.SourceChain
+{
+    public interface IRetrievalTypeConfirmations
+    {
+        int GetDepositConfirmations(int depositHeight, DepositRetrievalType retrievalType);
+
+        int GetDepositMaturityHeight(int depositHeight, DepositRetrievalType retrievalType);
+
+        /// <summary>
+        /// Determines the maximum distance to look back to find deposits that would be maturing at the specified height.
+        /// </summary>
+        /// <param name="maturityHeight">The minimum height at which deposits would be maturing.</param>
+        /// <returns>The maximum distance to look back to find deposits that would be maturing at the specified height.</returns>
+        int MaximumConfirmationsAtMaturityHeight(int maturityHeight);
+
+        DepositRetrievalType[] GetRetrievalTypes();
+    }
+
+    public class RetrievalTypeConfirmations : IRetrievalTypeConfirmations
+    {
+        private const int LowConfirmations = 10;         //  450 seconds - 7m30s
+        private const int MediumConfirmations = 25;      // 1125 seconds - 18m45s
+        private const int HighConfirmations = 50;        // 2250 seconds - 37m30s
+        private const int CirrusLowConfirmations = 30;   //  480 seconds - 8m0s
+        private const int CirrusMediumConfirmation = 70; // 1120 seconds - 18m40s
+        private const int CirrusHighConfirmations = 140; // 2240 seconds - 37m20s
+
+        private readonly NodeDeployments nodeDeployments;
+        private readonly Dictionary<DepositRetrievalType, int> legacyRetrievalTypeConfirmations;
+        private readonly Dictionary<DepositRetrievalType, int> retrievalTypeConfirmations;
+        private readonly Network network;
+
+        public RetrievalTypeConfirmations(Network network, NodeDeployments nodeDeployments, IFederatedPegSettings federatedPegSettings)
+        {
+            this.nodeDeployments = nodeDeployments;
+            this.network = network;
+            this.legacyRetrievalTypeConfirmations = new Dictionary<DepositRetrievalType, int>
+            {
+                [DepositRetrievalType.Small] = federatedPegSettings.MinimumConfirmationsSmallDeposits,
```

Comment:
We can actually remove these from the `federatedPegSettings` class as they will be obsolete after activation anyway,

Reply:
I was thinking we could postpone clean-up to a follow-up PR as there will be constants etc. related to legacy that may no longer be needed following activation.